core/trino-main/src/main/java/io/trino/node/InternalNode.java

```diff
@@ -41,15 +41,22 @@ public class InternalNode
     private final URI internalUri;
     private final NodeVersion nodeVersion;
     private final boolean coordinator;
+    private final boolean worker;
     private final long longHashCode;
 
     public InternalNode(String nodeIdentifier, URI internalUri, NodeVersion nodeVersion, boolean coordinator)
+    {
+        this(nodeIdentifier, internalUri, nodeVersion, coordinator, !coordinator);
+    }
+
+    public InternalNode(String nodeIdentifier, URI internalUri, NodeVersion nodeVersion, boolean coordinator, boolean worker)
```

Comment:
can a node be neither coordinator nor worker?
can a node be both coordinator and worker?

(`node-scheduler.include-coordinator` doesn't mean the coordinator needs to be called "a worker", even though it does work)

Reply:
we definitely don't want to keep two constructors.
i don't know whether we need a new one (per above), but **IF** we do, the old one should be deprecated or perhaps simply removed. this is internal code.

Reply:
Sad part is that the whole "the node can be either coordinator and worker" is for a single node setup, in which one is most likely not very much interested in this metric at all. However, I don't really like returning the worker count as nodecount - 1